### PR TITLE
SctRunner ➡️ AST

### PR DESCRIPTION
**Pull request by Tinggaard**
AST is awesome. Filenames are not.

### CODE REVIEW

**Comment by Tinggaard** [comment]
Pls squash btw

File: SocietalConstructionTool/SctRunner.cs [hunk 1]
```diff
@@ -43,12 +43,39 @@ private static SctParser GetSctParser(string input)
...
+            }
+            return newAstList;
+        }
+
+        private static SctProgramSyntax GetMergedAst(IEnumerable<SctProgramSyntax> astList, string rootNode)
+        {
+            IEnumerable<SctClassSyntax> classes = astList.SelectMany(ast => ast.Classes);
+            IEnumerable<SctFunctionSyntax> functions = astList.SelectMany(ast => ast.Functions);
+            var context = GetParser(rootNode).start();
+            return new SctProgramSyntax(context, functions, classes);
```
Comment (TheMagzuz) [inline SocietalConstructionTool/SctRunner.cs:73]: `context`->`astList.First().Context.OriginalContext`
Så kan rootNode argument fjernes

File: SocietalConstructionToolTests/Snapshots/ParserTests/BuiltinFunctions.verified.txt [hunk 2]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+﻿(start (class_def class Foo ( args_def ) { (class_body (state state Bar { (statement_list (statement (enter enter Bar <missing ';'>))) })) }) (function function f ( args_def ) -> (type void) { (statement_list (statement (expression_statement (call_expression exists ( (args_call (expression (agent_predicate Foo :: Bar ( args_agent )))) ))) ;) (statement (expression_statement (call_expression count ( (args_call (expression (agent_predicate Foo :: Bar ( args_agent )))) ))) ;) (statement (expression_statement (call_expression rand ( args_call ))) ;) (statement (expression_statement (call_expression seed ( (args_call (expression (literal 123))) ))) ;)) }) <EOF>)
```
Comment (TheMagzuz) [inline SocietalConstructionToolTests/Snapshots/ParserTests/BuiltinFunctions.verified.txt:1]: Fix syntax error in test file

File: SocietalConstructionToolTests/TestFiles/Parser/BuiltinFunctions.sct [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class Foo() {
     state Bar {
-
+        enter Bar
```
Comment (TheMagzuz) [inline SocietalConstructionToolTests/TestFiles/Parser/BuiltinFunctions.sct:3]: Semi-colon here

**Review by TheMagzuz — CHANGES_REQUESTED** [review]

File: SocietalConstructionToolTests/Snapshots/ParserTests/BuiltinFunctions.verified.txt (hunk 2 above)
Comment (Tinggaard) [reply SocietalConstructionToolTests/Snapshots/ParserTests/BuiltinFunctions.verified.txt:1]: image

**Review by TheMagzuz — APPROVED** [review]

**Pull request merged**